Configurations.md

```diff
@@ -2063,6 +2063,52 @@ fn lorem<T:Eq>(t:T) {
 
 See also: [`space_before_colon`](#space_before_colon).
 
+## `space_after_function_name`
+
+Leave a space after the function name.
+
+- **Default value**: `"Never"`
+- **Possible values**: `"AfterGenerics"`, `"Never"`
```

Comment:
I'm not convinced yet that a single config option with many variants would be the best way to handle all the corresponding scenarios.

Opening this door creates several spacing options that I see:

* no spaces period between the ident and opening paren
* space between the ident and opening paren w/no generics
* space between ident and generics and generics and paren
* lhs space on the generics but none on rhs
* rhs space on the generics but none on the lhs

Users would then also be able to argue that they should be able to have various combinations of these stylings, for example no spacing between the ident and paren in the absence of generics, but lhs, rhs, or both spacing in the presence of generics.

I understand you may prefer to use only one of those personally, but if we're opening up control of spacing between the ident and parens at all then we'll be forced to support the other minor spacing variants/combinations as well. 

If you still feel that this is best done as a single, multi-variant config option then please update this as described above to account for all such variants and combinations to see if that approach is really viable. Otherwise please change this to be a boolean that _only_ impacts rhs spacing in the presence of generics because `space_after_function_name = AfterGenerics`  is really unintuitive for the non-generic scenarios

Reply:
Using a boolean would make it easier to support those other styles later, except for "space between the ident and opening paren w/no generics". Should I make `space_before_fn_params` an enum of `"Always"`, `"Generics"`, and `"Never"` to support that style? Since it wasn't mentioned in the tracking issue, I think it's enough to keep the option as a boolean.

Reply:
> Should I make space_before_fn_params an enum of "Always", "Generics", and "Never" to support that style? Since it wasn't mentioned in the tracking issue, I think it's enough to keep the option as a boolean.

I probably need to think on this one a bit, but my initial reaction is that a boolean is probably fine. The no-generics, just a space between the name and paren scenario really is binary in nature so a boolean feels natural. It's the generics scenario that has the Never/OnlyBefore/OnlyAfter/BeforeAndAfter mix that needs the extra control. 

Think one boolean option for the `fn foo()` vs `fn foo ()` scenario and an enum option for `fn foo<T>()` vs `fn foo<T> ()` vs etc. should cover everything, including the two explicitly requested spacing styles

Reply:
All styles are covered now. Most of the code is needed to support adding a space before generics, since that style is inconsistent with other formatting for generics.